source/patterns/@aws-solutions-constructs/aws-wafwebacl-cloudfront/lib/index.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+// Imports
+import * as cloudfront from '@aws-cdk/aws-cloudfront';
+import * as waf from '@aws-cdk/aws-wafv2';
+import * as defaults from '@aws-solutions-constructs/core';
+// Note: To ensure CDKv2 compatibility, keep the import statement for Construct separate
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * @summary The properties for the WafwebaclToCloudFront class.
+ */
+export interface WafwebaclToCloudFrontProps {
+  /**
```

Comment:
Let's expand this comment to read:

/*
 * The existing CloudFront instance that will be protected with the WAF web ACL.
 *
 * This construct changes the CloudFront distribution by directly manipulating
 * the CloudFormation output, so this must be the Construct and cannot be 
 * changed to the Interface (IDistribution)
 * /

(That will save us inevitable conversation and research in the future)